src/Xml/XmlRuntime.fs

```diff
@@ -208,7 +208,7 @@ type XmlRuntime =
             | _ ->
                 match v with
                 | :? string         as v -> v
-                | :? DateTime       as v -> v.ToString("O", cultureInfo)
+                | :? DateTime       as v -> v.ToString("yyyy-MM-dd")
```

Comment:
Why this change?

Reply:
If we make `System.DateTime` correspond to `xs:date` (and `System.DateTimeOffset` correspond to `xs:dateTime`) when formatting a `xs:date` we have to avoid the time component in order to produce valid xml according to the XML Schema specification. On the other hand, when formatting a `xs:dateTime` it's ok to use the `"O"` format that is compliant with the W3C specs for `xs:dateTime`

Reply:
The problem is, even if we emit zeros for the time component, the resulting xml would be invalid.

Maybe we could use the "yyyy-MM-dd" format when the time component is zero and use the "O" format otherwise.

Reply:
That would work